framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceImplementation.java

```diff
@@ -254,15 +262,84 @@ public void updateContracts(
         inferredType = convertCFAbstractValueToAnnotatedTypeMirror(v, fieldDeclType);
         atypeFactory.wpiAdjustForUpdateNonField(inferredType);
       } else {
-        // This field is not in the store. Add its declared type.
-        inferredType = atypeFactory.getAnnotatedType(fieldElement);
+        // This field is not in the store. Use the declared type.
+        inferredType = fieldDeclType;
       }
       T preOrPostConditionAnnos =
-          storage.getPreOrPostconditionsForField(preOrPost, methodElt, fieldElement, atypeFactory);
+          storage.getPreOrPostconditions(
+              preOrPost, methodElt, fa.toString(), fieldDeclType, atypeFactory);
       String file = storage.getFileForElement(methodElt);
       updateAnnotationSet(
```

Comment:
`updateAnnotationSet` is called on the output of `getPreOrPostconditions`.

1. Why does that type-check?  `updateAnnotationSet`'s first type parameter is a `T`, but `getPreOrPostconditions` returns a `@Nullable T`.
2. I think `updateAnnotationSet`, and anywhere else that the result of `updateAnnotationSet` may flow, should have an early exit if the argument is null.

Reply:
> 1. Why does that type-check? 

I don't think it should. Is the Nullness Checker running on this code? I don't know if we run the Nullness Checker on all parts of the CF's codebase.

> 2. I think updateAnnotationSet, and anywhere else that the result of updateAnnotationSet may flow, should have an early exit if the argument is null

It seems to me that it would make more sense here to do a null-check on the result of `getPreOrPostConditions` and then skip the call to `updateAnnotationSet` if that result was null, because in that case there is nothing to update.

Reply:
On that second point, the code that I added for parameters and the receiver does in fact do a null check before calling `updateAnnotationSet`, so it should be safe. It's only the code for fields that needs to be changed. I added the null-check at this point, but if you feel strongly that we should change the spec of `updateAnnotationSet` instead I'm fine with doing that, instead.